src/raytracer/xmlreader.m

```diff
@@ -63,114 +65,149 @@
+            if materialSwitch==1
+                switch2=0;
+                for iterateMaterials=1:size(MaterialLibrary,1)
+                    if strcmpi(MaterialLibrary.Reflector{iterateMaterials},Material)
+                        CADOutputTemp(14) = MaterialLibrary.PrimaryKey(iterateMaterials);
+                        switch2=1;
+                    end
+                end
+
+                % Storing triangle vertices and plane equations in output
+                % Part where output file is created. It contains the triangle vertices
+                % in first nine columns, plane equations in the next four columns
+                if switch2==0
+                    materialSwitch=0;
                 end
+
+            end
+
+            CADOutputTemp(1:3) = round(v1,6);
+            CADOutputTemp(4:6) = round(v2,6);
+            CADOutputTemp(7:9) = round(v3,6);
+            CADOutputTemp(10:12) = round(normal,4);
+            CADOutputTemp(13) = round(D,4);
+
+            % We are using distance limitation at this step
+            if r==0
+                [switchDistance] = 1;
+            else
+                [switchDistance] = verifydistance(r,referencePoint,CADOutputTemp,1);
             end
-            
-            % Storing triangle vertices and plane equations in output
-            % Part where output file is created. It contains the triangle vertices
-            % in first nine columns, plane equations in the next four columns
-            if switch2==0
-                materialSwitch=0;
+
+            % If the triangles are within the given distance we increase the count,
+            % else the next triangle will replace the present row (as count remains constant)
+            if switchDistance==1
+%                 countRows=countRows+1;
+               CADOutput = [CADOutput;CADOutputTemp];
             end
-            
-        end
-        
-        CADOutput(indexCADOutput,1:3) = round(v1,6);
-        CADOutput(indexCADOutput,4:6) = round(v2,6);
-        CADOutput(indexCADOutput,7:9) = round(v3,6);
-        CADOutput(indexCADOutput,10:12) = round(normal,4);
-        CADOutput(indexCADOutput,13) = round(D,4);
-        
-        % We are using distance limitation at this step
-        if r==0
-            [switchDistance] = 1;
-        else
-            [switchDistance] = verifydistance(r,referencePoint,CADOutput,indexCADOutput);
-        end
-        
-        % If the triangles are within the given distance we increase the count,
-        % else the next triangle will replace the present row (as count remains constant)
-        if switchDistance==1
-            countRows=countRows+1;
+
         end
-        
     end
 end
-
 end
 
 
 %% Utils
-function v = getTriangleVertex(s,volumeIdx,triangIdx,vertexIdx,sizeVolume)
+function v = getTriangleVertex(stemp,volumeIdx,triangIdx,vertexIdx,sizeVolume)                      
```

Comment:
again, does stemp mean something? i don't find it as a real word: https://www.thesaurus.com/misspelling?term=stemp&s=t

Reply:
No I think I wanted to use a temporary variable for s that why I used stamp. I can change it no issue